src/main.rs

```diff
@@ -277,17 +279,40 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
             } else {
                 ingredient_count.to_string()
             };
+            let source_msg = match *ingredient_source {
+                crafting::Source::TradingPost => if ingredient.max_price == ingredient.min_price {
+                    format!(
+                        " (at {}) Subtotal: {}",
+                        copper_to_string(ingredient.min_price),
+                        copper_to_string(ingredient.total_cost.to_integer()),
+                    )
+                } else {
+                    format!(
+                        " (at {} to {}) Subtotal: {}",
+                        copper_to_string(ingredient.min_price),
+                        copper_to_string(ingredient.max_price),
+                        copper_to_string(ingredient.total_cost.to_integer()),
+                    )
+                },
+                crafting::Source::Vendor => {
+                    let cost = items_map.get(ingredient_id).unwrap_or_else(|| {
+                        panic!("Missing item for ingredient {}", ingredient_id)
+                    }).vendor_cost().unwrap_or(0);
```

Comment:
Would it make sense to not show the cost if it's missing?

Reply:
I've left the `panic!` in place from the code I moved around. I agree with the author of that section of code, that it should be vanishingly unlikely that we wouldn't find an item in list of items from the API. If we do, the best response is probably to break instead of recovering and behaving wrong.

If you're referring to the `unwrap_or(0)` - I agree. Change committed and pushed.